Emit an error if Kani executable was not found

zhassan-aws opened the pull request:
### Description of changes: 

Check if `which cargo-kani` returned an error, abort execution, and set the test result as "errored". This is what gets shown in VSCode:

![image](https://github.com/model-checking/kani-vscode-extension/assets/88045115/eac35344-b1e1-41da-9d1f-42e3333871fb)

### Resolved issues:

Resolves #72 

### Related RFC:

<!--
Link to the Tracking RFC issue if this work implements part of an RFC.
-->
Optional #ISSUE-NUMBER.

### Call-outs:

<!-- 
Address any potentially confusing code. Is there code added that needs to be cleaned up later? Is there code that is missing because it’s still in development? 
-->

A cleaner approach would be to check on load if Kani is installed, and if not put the extension in an error state and prevent running any harnesses.

### Testing:

* How is this change tested? Removed kani from path, launched extension, and ran one of the harnesses.

* Is this a refactor change?

### Checklist
- [X] Each commit message has a non-empty body, explaining why the change was made
- [X] Methods or procedures are documented
- [X] Regression or unit tests are included, or existing tests cover the modified code
- [X] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

jaisnan reviewed (commented):
The changes look great! I do think the cleaner solution can be added easily as well on top of these changes. It should be simple to implement as we already do exactly the same, with cargo here - https://github.com/model-checking/kani-vscode-extension/blob/15c27c756b029833d126d12b4032b9c5aa1e6175/src/extension.ts#LL34C3-L34C3.

jaisnan commented:
Could I ask if you could add a screenshot of what the end result looks like visually? And also what underlying environment conditions allowed you to trigger the issue (i.e what was the cargo-kani path in your case). Any information on how it was tested would be helpful to me.

Thanks!

zhassan-aws commented:
> Could I ask if you could add a screenshot of what the end result looks like visually? And also what underlying environment conditions allowed you to trigger the issue (i.e what was the cargo-kani path in your case). Any information on how it was tested would be helpful to me.

I added a script to the PR description as well as how I tested it. In my case, I removed kani from the PATH environment variable, which causes `which cargo-kani` to exit with an error status.

jaisnan commented:
> > Could I ask if you could add a screenshot of what the end result looks like visually? And also what underlying environment conditions allowed you to trigger the issue (i.e what was the cargo-kani path in your case). Any information on how it was tested would be helpful to me.
> 
> I added a script to the PR description as well as how I tested it. In my case, I removed kani from the PATH environment variable, which causes `which cargo-kani` to exit with an error status.

Thanks, that should be good enough to work. The final change looks great too.

zhassan-aws commented:
> I do think the cleaner solution can be added easily as well on top of these changes. It should be simple to implement as we already do exactly the same, with cargo here

I couldn't get this approach to work. My approach is in [this branch](https://github.com/model-checking/kani-vscode-extension/compare/main...zhassan-aws:kani-vscode-extension:iss72-alt). With those changes, an error is popped-up as the extension was loading:

![image](https://github.com/model-checking/kani-vscode-extension/assets/88045115/632b9858-aaf6-4575-b4f6-37f2e053e7e0)

but the extension still activated and running any harness leads causes the wheel to keep spinning forever.

jaisnan reviewed (approved):
Thanks!
